packages/studio-web/src/app/upload/upload.component.ts

```diff
@@ -85,7 +86,10 @@ export class UploadComponent implements OnInit {
       langs: this.rasService.getLangs$(),
       _: this.ssjsService.waitForInit$(),
     })
-      .pipe(finalize(() => (this.isLoaded = true)))
+      .pipe(
+        finalize(() => (this.isLoaded = true)),
```

Comment:
I'm a bit confused by this - if `finalize()` was already working here (it was), doesn't that mean that the Observable completed successfully and the subscription ended?

Do you have a reference for this pattern and the reason it's necessary?  I'm not totally convinced...

Reply:
I think this is because of forkJoin which completes after the first emission. So, you're right that it's not necessary here. as a rule of thumb, I just apply this pattern to each subscription because it doesn't hurt and because applying it selectively would require analyzing each subscription as to whether there's a `take(1)` buried in the service somewhere or knowing that `forkJoin` completes after a single emission etc. I can try and dig up some documentation on the pattern

Reply:
I see what you mean, it doesn't hurt or complicate the code unnecessarily, and covers the case where, for whatever reason, the component gets destroyed when a subscription is still outstanding...

Reply:
Yea, it looks like the only place this might happen is when we call `this.rasService.convertRasFormat$` - everything else either has `forkJoin` or `take(1)` applied to it seemingly. But, ya, as a rule of thumb I don't think it's a bad pattern to explicitly unsubscribe, and it gives us an easy way of doing so in the future if we introduce more observables that don't complete on their own.